cnvlib/call.py

```diff
@@ -237,16 +231,22 @@ def _reference_expect_copies(chrom, ploidy, is_sample_female, is_reference_male)
         A pair of integers: number of copies in the reference, and expected in
         the sample.
     """
-    chrom = chrom.lower()
-    if chrom in ["chrx", "x"]:
-        ref_copies = ploidy // 2 if is_reference_male else ploidy
-        exp_copies = ploidy if is_sample_female else ploidy // 2
-    elif chrom in ["chry", "y"]:
-        ref_copies = ploidy // 2
-        exp_copies = 0 if is_sample_female else ploidy // 2
-    else:
-        ref_copies = exp_copies = ploidy
-    return ref_copies, exp_copies
+    df = cnarr.copy().data
+
+    # Set all to default value (i.e. for autsosomes):
+    df["reference"] = np.repeat(ploidy, len(df))
+    df["expect"] = np.repeat(ploidy, len(df))
+
+    df.loc[df.chromosome == cnarr.chr_x_label, "reference"] = (
+        ploidy // 2 if is_reference_male else ploidy
+    )
+    df.loc[df.chromosome == cnarr.chr_x_label, "expect"] = (
+        ploidy if is_sample_female else ploidy // 2
+    )
+
+    df.loc[df.chromosome == cnarr._chr_y_label, "reference"] = ploidy // 2
```

Comment:
The better fix here would be to change the parameter name `is_reference_male` to `is_haploid_x_reference` or equivalent. I changed the CLI option name from `--male-reference` to `--haploid-x-reference` in commands.py but it looks like I didn't bother renaming arguments in the deeper function signatures.

The reference profiles come in two flavors: XY (formerly "male" reference, now "haploid x") and XXY (never documented as "female"; it's just the default):
https://cnvkit.readthedocs.io/en/stable/sex.html#reference-sex-chromosome-ploidy

Why does the reference profile have Klinefelter syndrome by default? So that male samples' chrY values can be calculated, while plots still look reasonable (for female samples, copy-neutral X will look the same as the autosomes and there won't be any signal for Y). If reference chrY is 0 then we get divide-by-zero problems.

Reply:
Okay, I replaced occurrences of `reference_male` by `is_haploid_x_reference` (except occurrences of `args.male_reference`) and added a comment.